zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -336,88 +345,82 @@ mod test {
         eth_types::Transaction {
             from,
             to: Some(to),
-            value: value.unwrap_or(point_one_ether),
-            gas: gas.map(Word::from).unwrap_or(minimal_gas),
-            gas_price: gas_price.or(Some(two_gwei)),
+            value,
+            gas,
+            gas_price: Some(gas_price),
             input: calldata.into(),
             ..Default::default()
         }
     }
 
     #[test]
     fn begin_tx_gadget_simple() {
-        // Transfer 1 ether, successfully
-        test_ok(mock_tx(None, None, None, vec![]), true);
+        // Transfer 0.1 ether, successfully
+        test_ok(
+            mock_tx(*POINT_ONE_ETHER, *MINIMAL_GAS, *TWO_GWEI, vec![]),
+            true,
+        );
 
-        // Transfer 1 ether, tx reverts
-        test_ok(mock_tx(None, None, None, vec![]), false);
+        // Transfer 0.1 ether, tx reverts
+        test_ok(
+            mock_tx(*POINT_ONE_ETHER, *MINIMAL_GAS, *TWO_GWEI, vec![]),
+            false,
+        );
 
         // Transfer nothing with some calldata
         test_ok(
-            mock_tx(None, Some(21086), None, vec![1, 2, 3, 4, 0, 0, 0, 0]),
+            mock_tx(
+                Word::zero(),
+                Word::from(21086),
+                *TWO_GWEI,
+                vec![1, 2, 3, 4, 0, 0, 0, 0],
+            ),
             false,
         );
     }
 
     #[test]
     fn begin_tx_gadget_rand() {
-        let point_one_ether = Word::from(10u8).pow(Word::from(17u8));
+        let random_amount = Word::from_little_endian(&rand_bytes(32)) % *POINT_ONE_ETHER;
+
+        let max_gas_price = 476_054_460_630_296u64;
+        assert_eq!(Word::from(max_gas_price), *TEN_ETHER / *MINIMAL_GAS);
+        let random_gas_price = Word::from(rand_range(0..max_gas_price));
+
+        // If this test fails, we want these values to appear in the CI logs.
+        dbg!(random_amount, random_gas_price);
 
         // Transfer random ether, successfully
         test_ok(
-            mock_tx(
-                Some(Word::from_little_endian(&rand_bytes(32)) % point_one_ether),
-                None,
-                None,
-                vec![],
-            ),
+            mock_tx(random_amount, *MINIMAL_GAS, *TWO_GWEI, vec![]),
             true,
         );
 
         // Transfer nothing with random gas_price, successfully
         test_ok(
-            mock_tx(
-                None,
-                None,
-                Some(Word::from(rand_range(0..47619047619047u64))),
-                vec![],
-            ),
+            mock_tx(Word::zero(), *MINIMAL_GAS, random_gas_price, vec![]),
             true,
         );
 
         // Transfer random ether, tx reverts
         test_ok(
-            mock_tx(
-                Some(Word::from_little_endian(&rand_bytes(32)) % point_one_ether),
-                None,
-                None,
-                vec![],
-            ),
+            mock_tx(random_amount, *MINIMAL_GAS, *TWO_GWEI, vec![]),
```

Comment:
mock_tx could keep recieving Options and use these consts as fallback for Default.

Reply:
I made this more explicit because previously, when we were using None to represent default, the code did not do what the comments claimed. E.g. `mock_tx(Word::zero(), ...)` would not transfer 0, but would transfer 0.1 eth instead.

Reply:
I changed this to be more explicit because previously the comments were saying that passing `amount=None` would transfer 0 eth when in fact the default value was 0.1 eth. This way it's harder to make that mistake.